fdbclient/MasterProxyInterface.h

```diff
@@ -104,14 +104,15 @@ struct CommitID {
 	Version version; 			// returns invalidVersion if transaction conflicts
 	uint16_t txnBatchId;
 	Optional<Value> metadataVersion;
+	Optional<Standalone<VectorRef<KeyRangeRef>>> conflictingKeyRanges;
```

Comment:
Sorry for this rather late suggestion: I think a lot of your code would become cleaner if this is a `ReadYourWritesTransaction`. It would also be faster as then you would only need to add all ranges to it once